Fabric.Authorization.Client/AuthorizationClient.cs

```diff
@@ -35,5 +283,26 @@ public AuthorizationClient(HttpClient client)
             var error = JsonConvert.DeserializeObject<Error>(stringResponse);
             throw new AuthorizationException(error);
         }
+
+        private async Task SendRequest(HttpRequestMessage message)
+        {
+            var response = await client.SendAsync(message).ConfigureAwait(false);
+            var stringResponse = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                if (response.StatusCode < HttpStatusCode.InternalServerError)
+                {
+                    var error = JsonConvert.DeserializeObject<Error>(stringResponse);
+                    throw new AuthorizationException(error);
+                }
+
+                throw new AuthorizationException(new Error
+                {
+                    Code = response.StatusCode.ToString(),
+                    Message = "Unknown error occurred"

```

Comment:
Not intending to block this PR, but we should probably either return the content of the response in the `Message`, or have the message set to "Internal Server Error".